backend/exchanges/exchanges.go

```diff
@@ -0,0 +1,82 @@
+var regionCodes = [...]string{
+	"AD", "AE", "AF", "AG", "AI", "AL", "AM", "AO", "AQ", "AR", "AS", "AT", "AU", "AW", "AX", "AZ",
+	"BA", "BB", "BD", "BE", "BF", "BG", "BH", "BI", "BJ", "BL", "BM", "BN", "BO", "BQ", "BR", "BS",
+	"BT", "BV", "BW", "BY", "BZ", "CA", "CC", "CD", "CF", "CG", "CH", "CI", "CK", "CL", "CM", "CN",
+	"CO", "CR", "CU", "CV", "CW", "CX", "CY", "CZ", "DE", "DJ", "DK", "DM", "DO", "DZ", "EC", "EE",
+	"EG", "EH", "ER", "ES", "ET", "FI", "FJ", "FK", "FM", "FO", "FR", "GA", "GB", "GD", "GE", "GF",
+	"GG", "GH", "GI", "GL", "GM", "GN", "GP", "GQ", "GR", "GS", "GT", "GU", "GW", "GY", "HK", "HM",
+	"HN", "HR", "HT", "HU", "ID", "IE", "IL", "IM", "IN", "IO", "IQ", "IR", "IS", "IT", "JE", "JM",
+	"JO", "JP", "KE", "KG", "KH", "KI", "KM", "KN", "KP", "KR", "KW", "KY", "KZ", "LA", "LB", "LC",
+	"LI", "LK", "LR", "LS", "LT", "LU", "LV", "LY", "MA", "MC", "MD", "ME", "MF", "MG", "MH", "MK",
+	"ML", "MM", "MN", "MO", "MP", "MQ", "MR", "MS", "MT", "MU", "MV", "MW", "MX", "MY", "MZ", "NA",
+	"NC", "NE", "NF", "NG", "NI", "NL", "NO", "NP", "NR", "NU", "NZ", "OM", "PA", "PE", "PF", "PG",
+	"PH", "PK", "PL", "PM", "PN", "PR", "PS", "PT", "PW", "PY", "QA", "RE", "RO", "RS", "RU", "RW",
+	"SA", "SB", "SC", "SD", "SE", "SG", "SH", "SI", "SJ", "SK", "SL", "SM", "SN", "SO", "SR", "SS",
+	"ST", "SV", "SX", "SY", "SZ", "TC", "TD", "TF", "TG", "TH", "TJ", "TK", "TL", "TM", "TN", "TO",
+	"TR", "TT", "TV", "TW", "TZ", "UA", "UG", "UM", "US", "UY", "UZ", "VA", "VC", "VE", "VG", "VI",
+	"VN", "VU", "WF", "WS", "YE", "YT", "ZA", "ZM", "ZW"}
+
+// ExchangeRegionList contains a list of ExchangeRegion objects.
+type ExchangeRegionList struct {
+	Regions []ExchangeRegion `json:"regions"`
+}
+
+// ExchangeRegion contains the ISO 3166-1 alpha-2 code of a specific region and a boolean
+//for each exchange, indicating if that exchange is enabled for the region.
+type ExchangeRegion struct {
+	Code             string `json:"code"`
+	IsMoonpayEnabled bool   `json:"isMoonpayEnabled"`
+	IsPocketEnabled  bool   `json:"isPocketEnabled"`
+}
+
+// PaymentMethod type is used for payment options in exchange deals.
+type PaymentMethod string
+
+const (
+	// CardPayment is a payment with credit/debit card.
+	CardPayment PaymentMethod = "card"
+	// BankTransferPayment is a payment with bank transfer.
+	BankTransferPayment PaymentMethod = "bank-transfer"
+)
+
+// ExchangeDeal represents a specific purchase option of an exchange.
+type ExchangeDeal struct {
+	Fee     float32       `json:"fee"`
+	Payment PaymentMethod `json:"payment"`
+	IsFast  bool          `json:"isFast"`
+}
+
+// ListExchangesByRegion populates an array of `ExchangeRegion` objects representing the availability
+// of the various exchanges in each of them.
+func ListExchangesByRegion(httpClient *http.Client) (ExchangeRegionList, error) {
+	moonpayRegions, err := GetMoonpaySupportedRegions(httpClient)
+	if err != nil {
+		return ExchangeRegionList{}, err
```

Comment:
here and below: it would not be good if the buy page couldn't load just because one of the two api endpoints is not responding correctly. is that the case now? if so, let's think about graceful failures.

would be especially bad if one exchange endpoint failing would also stop the other from being accessible, but also the buy widget could work even if the countries api does not respond correctly.